### ISSUE TITLE
MD005 not respecting MD007 ident configuration in rumdl 0.0.126

### ISSUE THREAD

**Issue by owenlamont** [rvben/rumdl#64]
I had rule MD005 and MD007 enabled and had configured MD007 with a 4 space indent, e.g.

```toml
[tool.rumdl.MD007]
# MD007/unordered-list-indent
indent = 4
```

But when MD005 ran with fix, it unindented all the 4 space indented sub lists, e.g.

<img width="1905" height="441" alt="Image" src="https://github.com/user-attachments/assets/e7907c20-56b3-4c8d-a568-d33e8d191e05" />

This behaviour isn't consistent with markdownlint-cli (ruff itself uses markdownlint-cli so I thought it would be a good test to try substituting rumdl into its pre-commit tools).

P.S. when doing the automatic migration from the `.markdownlint.yaml` in the ruff repo - several rules they outright disabled did not get disabled in the generated .rumdl.toml - this was their `.markdownlint.yaml`:

```yaml
# default to true for all rules
default: true

# MD007/unordered-list-indent
MD007:
  indent: 4

# MD033/no-inline-html
MD033: false

# MD041/first-line-h1
MD041: false

# MD013/line-length
MD013: false

# MD014/commands-show-output
MD014: false

# MD024/no-duplicate-heading
MD024:
  # Allow when nested under different parents e.g. CHANGELOG.md
  siblings_only: true

# MD046/code-block-style
#
# Ignore this because it conflicts with the code block style used in content
# tabs of mkdocs-material which is to add a blank line after the content title.
#
# Ref: https://github.com/astral-sh/ruff/pull/15011#issuecomment-2544790854
MD046: false

# Link text should be descriptive
# Disallows link text like *here* which is annoying.
MD059: false
```

and this was the generated `.rumdl.toml`:

```toml
[global]
disable = ["MD046"]

[MD007]
indent = 4

[MD024]
siblings-only = true
```

**Comment by rvben** [comment]
Hi @owenlamont,

Thanks for raising this issue. This was not correct indeed, and the indentation was hardcoded directly for MD005 so that conflicted with MD007. This is now resolved for the upcoming release.

Thanks again. Issues like this really help everyone to have a better experience.

**Comment by owenlamont** [comment]
Hi @rvben  - you're welcome! Thanks for addressing these issues so fast. I did notice other discrepancies with markdownlint-cli in that test like how they disable MD041 but leave the other heading rules enabled - rumdl still enforced h1 headings and markdownlint didn't. It wasn't so clear that it was wrong behaviour though - just different... If you like, I can raise all the inconsistencies as issues for you to make a call on.

**Comment by rvben** [comment]
Hi @owenlamont,

If you want, I think it'd be very helpful to get more clarity on the inconsistencies. At the very least we'd have it documented in issues, but in a lot of cases it probably helps to improve. :)

**Comment by owenlamont** [comment]
Hi @rvben - I just tested again with rumdl v0.0.128 but I still get the same behaviour.

For reference I'm working on a recent fork of [ruff](https://github.com/astral-sh/ruff) - in my dev branch I deleted their `.markdownlint.yaml`

```yaml
# default to true for all rules
default: true

# MD007/unordered-list-indent
MD007:
  indent: 4

# MD033/no-inline-html
MD033: false

# MD041/first-line-h1
MD041: false

# MD013/line-length
MD013: false

# MD014/commands-show-output
MD014: false

# MD024/no-duplicate-heading
MD024:
  # Allow when nested under different parents e.g. CHANGELOG.md
  siblings_only: true

# MD046/code-block-style
#
# Ignore this because it conflicts with the code block style used in content
# tabs of mkdocs-material which is to add a blank line after the content title.
#
# Ref: https://github.com/astral-sh/ruff/pull/15011#issuecomment-2544790854
MD046: false

# Link text should be descriptive
# Disallows link text like *here* which is annoying.
MD059: false
```

Added the following to their `pyproject.toml`:

```toml
[tool.rumdl]
# See rules reference: https://github.com/rvben/rumdl/blob/main/docs/RULES.md

# Ignore MD013/line-length, MD014/commands-show-output, MD033/no-inline-html,
# MD041/first-line-h1, 
# MD046/code-block-style because it conflicts with the code block style used in 
# content tabs of mkdocs-material which is to add a blank line after the content title.
# MD059/Link text should be descriptive
# Disallows link text like *here* which is annoying.
#
# Ref: https://github.com/astral-sh/ruff/pull/15011#issuecomment-2544790854
disable = ["MD013", "MD014", "MD033", "MD041", "MD046", "MD059"]

[tool.rumdl.MD007]
# MD007/unordered-list-indent
indent = 4

[tool.rumdl.MD024]
# Allow when nested under different parents e.g. CHANGELOG.md
siblings-only = true
```

and substituted their old markdown-lint pre-commit reference:

```yaml
  - repo: https://github.com/igorshubovych/markdownlint-cli
    rev: v0.45.0
    hooks:
      - id: markdownlint-fix
        exclude: |
          (?x)^(
            docs/formatter/black\.md
            | docs/\w+\.md
          )$
```

with

```yaml
  - repo: https://github.com/rvben/rumdl-pre-commit
    rev: v0.0.128
    hooks:
      - id: rumdl
        args: [--fix]
        exclude: |
          (?x)^(
            docs/formatter/black\.md
            | docs/\w+\.md
          )$
```

but this still happened when running `pre-commit run --all-files`

<img width="1739" height="520" alt="Image" src="https://github.com/user-attachments/assets/f58d59d5-1f11-4601-9729-4e7883444be5" />
